### PR TITLE
Changed frontend to use axios for stats component functions

### PR DESCRIPTION
**Pull request by AkashAryal**
### Summary 

This pull request updates the front end to use axios to call the new express endpoints for functions **totalReviews**, **howManyReviewsEachClass**, **howManyEachClass**, and **topSubjects**. These are the main non chart functions in the admin page.

### Test Plan 
Checked locations where these functions were used to ensure things still work.

### CODE REVIEW

File: server/endpoints/AdminChart.ts [hunk 1]
```diff
@@ -187,7 +187,7 @@ const topSubjectsCB = async (_ctx: Context) => {
  * Returns the top 15 subjects (in terms of number of reviews)
  */
 export const topSubjects: Endpoint<unknown> = {
-  guard: [body("token").notEmpty()],
+  guard: [],
```
Comment (khemritolya) [inline server/endpoints/AdminChart.ts:190]: Why did you remove the guard?

Comment (AkashAryal) [reply server/endpoints/AdminChart.ts:190]: When I first migrated the backend topSubjects method I thought it did take in an auth token (thus i put it in the guard) but it turns out that it does not take in anything (so i removed it in this PR). Should we alter the topSubjects method to require a token? (I think that makes sense too).

Comment (khemritolya) [reply server/endpoints/AdminChart.ts:190]: If we don't already, we should probably gate all the admin functions behind tokens.

**Review by khemritolya — COMMENTED** [review]
Looks good, but I tested a bit more, and I found that there seem to still be calls to the old topSubjects around, at least when I check console output. Can you confirm? Other ones seem to sucessfully be not longer called.

**Comment by AkashAryal** [comment]
> Looks good, but I tested a bit more, and I found that there seem to still be calls to the old topSubjects around, at least when I check console output. Can you confirm? Other ones seem to sucessfully be not longer called.

The method calling the old topSubjects is the big chart method. Once I update the chart function in the next pr, the issue should fix itself.

**Review by khemritolya — APPROVED** [review]
Looks good!
